lib/qdrant-docker-image-ecr-deployment-cdk-stack.ts

```diff
@@ -1,16 +1,26 @@
 import * as cdk from 'aws-cdk-lib';
 import { Construct } from 'constructs';
-// import * as sqs from 'aws-cdk-lib/aws-sqs';
+import * as ecrDeploy from 'cdk-ecr-deployment';
+import * as ecr from 'aws-cdk-lib/aws-ecr';
+import { QdrantDockerImageEcrDeploymentCdkStackProps } from './QdrantDockerImageEcrDeploymentCdkStackProps';
 
 export class QdrantDockerImageEcrDeploymentCdkStack extends cdk.Stack {
-  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
-    super(scope, id, props);
+    constructor(scope: Construct, id: string, props: QdrantDockerImageEcrDeploymentCdkStackProps) {
+        super(scope, id, props);
 
-    // The code that defines your stack goes here
+        const ecrRepository = new ecr.Repository(this, `${props.appName}-QdrantDockerImageEcrRepository`, {
+            repositoryName: props?.repositoryName ?? 'qdrant-docker-image-ecr-deployment-cdk',
+            removalPolicy: cdk.RemovalPolicy.DESTROY,
+            encryption: ecr.RepositoryEncryption.AES_256
+        });
 
-    // example resource
-    // const queue = new sqs.Queue(this, 'QdrantDockerImageEcrDeploymentCdkQueue', {
-    //   visibilityTimeout: cdk.Duration.seconds(300)
-    // });
-  }
+        ecrRepository.addLifecycleRule({ maxImageAge: cdk.Duration.days(7), rulePriority: 1, tagStatus: ecr.TagStatus.UNTAGGED }); // delete images older than 7 days
+        ecrRepository.addLifecycleRule({ maxImageCount: 4, rulePriority: 2, tagStatus: ecr.TagStatus.ANY }); // keep last 4 images
+
+        // Copy from docker registry to ECR.
+        new ecrDeploy.ECRDeployment(this, `${props.appName}-QdrantDockerImageEcrDeployment`, {
+            src: new ecrDeploy.DockerImageName('qdrant/qdrant:latest'),
+            dest: new ecrDeploy.DockerImageName(`${ecrRepository.repositoryUri}:${props.imageVersion}`),
```

Comment:
**Suggestion:** Use the `props.repositoryName` directly in the `DockerImageName` instead of hardcoding 'qdrant/qdrant:latest'.
```suggestion
        new ecrDeploy.ECRDeployment(this, `${props.appName}-QdrantDockerImageEcrDeployment`, {
            src: new ecrDeploy.DockerImageName(`${props.repositoryName}:${props.imageVersion}`),
            dest: new ecrDeploy.DockerImageName(`${ecrRepository.repositoryUri}:${props.imageVersion}`),
        });
```